### PR TITLE
System.Security: Consolidate IsLegalKeySize and related methods into common code.

### PR DESCRIPTION
**Pull request by filipnavara**

### CODE REVIEW

File: src/Common/src/System/Security/Cryptography/KeySizeHelpers.cs [hunk 1]
```diff
@@ -8,16 +8,31 @@
...
+    internal static class KeySizeHelpers
     {
-        public static byte[] CloneByteArray(this byte[] src)
+        public static KeySizes[] CloneKeySizesArray(this KeySizes[] src)
         {
-            if (src == null)
+            return (KeySizes[])(src.Clone());
+        }
+
+        public static bool UsesIv(this CipherMode cipherMode)
```
Comment (bartonjs) [inline src/Common/src/System/Security/Cryptography/KeySizeHelpers.cs]: This isn't a KeySizes operation.

File: src/System.Security.Cryptography.Primitives/src/Resources/Strings.resx [hunk 2]
```diff
@@ -106,6 +106,9 @@
   <data name="Cryptography_InvalidPaddingMode" xml:space="preserve">
     <value>Specified padding mode is not valid for this algorithm.</value>
   </data>
+  <data name="Cryptography_MissingIV" xml:space="preserve">
+    <value>The cipher mode specified requires that an initialization vector (IV) be used.</value>
```
Comment (bartonjs) [inline src/System.Security.Cryptography.Primitives/src/Resources/Strings.resx]: The resource strings that are being added suggest a bad "common factoring".

Comment (filipnavara) [reply src/System.Security.Cryptography.Primitives/src/Resources/Strings.resx]: Well, I can skip that one method. Will update the PR.

File: src/Common/src/System/Security/Cryptography/KeySizeHelpers.cs [hunk 3]
```diff
@@ -8,16 +8,11 @@
 
 namespace System.Security.Cryptography
 {
-    internal static class Helpers
+    internal static class KeySizeHelpers
     {
-        public static byte[] CloneByteArray(this byte[] src)
```
Comment (bartonjs) [inline src/Common/src/System/Security/Cryptography/KeySizeHelpers.cs]: Where did this CloneByteArray method go?  Is it just no longer used?

Comment (filipnavara) [reply src/Common/src/System/Security/Cryptography/KeySizeHelpers.cs]: Reused from Common/Internal/Security/Helpers.cs, which was already included in the other projects.

**Comment by bartonjs** [comment]
Looks like you'll have to rebase this one.

**Comment by filipnavara** [comment]
Rebased. Thanks for review!

**Pull request merged**